HIP-Basic/CMakeLists.txt

```diff
@@ -25,8 +25,45 @@ project(HIP-Basic LANGUAGES CXX)
 
 # Only supported on HIP (not CUDA)
 if(NOT "${GPU_RUNTIME}" STREQUAL "CUDA")
-    add_subdirectory(assembly_to_executable)
-    add_subdirectory(llvm_ir_to_executable)
+
+    # Make sure the dependencies can be found before building.
+    find_program(
+        LLVM_DIS_COMMAND llvm-dis
+        PATH_SUFFIXES bin
+        PATHS
+        ${ROCM_ROOT}/llvm
+        ${CMAKE_INSTALL_PREFIX}/llvm)
+    find_program(
+        OFFLOAD_BUNDLER_COMMAND clang-offload-bundler
+        PATH_SUFFIXES bin
+        PATHS
+        ${ROCM_ROOT}/llvm
+        ${CMAKE_INSTALL_PREFIX}/llvm)
+    find_program(
+        LLVM_MC_COMMAND llvm-mc
+        PATH_SUFFIXES bin
+        PATHS
+        ${ROCM_ROOT}/llvm
+        ${CMAKE_INSTALL_PREFIX}/llvm)
+
+    if(LLVM_DIS_COMMAND_FOUND AND OFFLOAD_BUNDLER_COMMAND_FOUND AND LLVM_MC_COMMAND_FOUND)
+        add_subdirectory(assembly_to_executable)
+    else()  
+        message("'llvm-dis' or 'llvm-mc' or 'clang-offload-bundler' not found, not building assembly example.")
```

Comment:
```suggestion
        message("'llvm-dis', 'llvm-mc', or 'clang-offload-bundler' not found, not building assembly example.")
```
It may also be nice to output which program is missing, but this is fine for now.

Reply:
Used commas 😃